### ISSUE TITLE
Imrpove format of synapse_review_recent_signups output to easier parsing as JSON or XML

### ISSUE THREAD

**Issue by MurzNN** [matrix-org/synapse#10434]
**Description:**
The recently added `synapse_review_recent_signups` is great, but my homeserver I have use bridges, so they constantly creates a lot of virtual accounts such as `@_telegram_123456:example.com` and `@_slackpuppet___t02_f_l_v1_a_n=2d_u027_h616_b41:example.com`, that shown as long list in output of `synapse_review_recent_signups` script, and I can't find the real registered users between that accounts.

And the output format is not so easy to parse, for example, via `grep`, `jq`, or any other cli tool:
```
# Created: 2021-07-13 18:55:08
# Email: None
# IPs: 
# Number joined public rooms: 0 
# Number joined private rooms: 1
#
@_slackpuppet___t02_f_l_v1_h_n=2d_u028_a9_e2_k_p_b:example.com

# Created: 2021-07-13 19:47:54
# Email: None
# IPs: 
# Number joined public rooms: 0 
# Number joined private rooms: 0
#
@_slackpuppet___t02_f_l_v1_h_n=2d_u027_h616_b47:example.com

# Created: 2021-07-13 20:17:02
# Email: None
# IPs: 
# Number joined public rooms: 0 
# Number joined private rooms: 1
#
@_telegram_1234567:example.com
```

To solve this problem will be good to add some parseable formatters to output, for example - JSON or XML format.

Alternative can be adding support for filtering account name by regex into the script, but this will add more complexity to it and anyway don't give all flexibility like custom output parsers.

**Comment by anoadragon453** [comment]
Generally bridges tend to place a `_` directly after the `@` in their matrix ID in order to differentiate bridged users from native Matrix users -- though this is not a requirement.

Agreed on the formatting, though I believe this script was initially designed solely for admins to use to check for spam bots, rather than something to be consumed by a further processing pipeline. For that purpose it may be a better idea to extend the [List Users Admin API endpoint](https://matrix-org.github.io/synapse/latest/admin_api/user_admin_api.html#list-accounts) to return (and allow sorting by) user creation time.

**Comment by babolivier** [comment]
> For that purpose it may be a better idea to extend the List Users Admin API endpoint to return (and allow sorting by) user creation time.

Wouldn't it be easier to add the appservice_id to the admin API and allow filtering on that?

**Comment by MurzNN** [comment]
> For that purpose it may be a better idea to extend the List Users Admin API endpoint to return (and allow sorting by) user creation time.

Yeah, extending the Admin API will be much better, than raw scripting! Also API now miss information about last activity (last login, last sent event) and IP. If implement this - seems that script can be reworked to work independently of Synapse via API calls only!

**Comment by MurzNN** [comment]
> Generally bridges tend to place a `_` directly after the `@` in their matrix ID in order to differentiate bridged users from native Matrix users -- though this is not a requirement.

Yes, also some setups have native users (bots), that have predictable names, so will be good to have regexp match filter in script and in API too...
